runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/GrpcWindmillServer.java

```diff
@@ -862,6 +870,161 @@ public final Instant startTime() {
     }
   }
 
+  static class GetWorkTimingInfosTracker {
+
+    private Instant workItemCreationEndTime = Instant.EPOCH;
+    private Instant workItemLastChunkReceivedByWorkerTime = Instant.EPOCH;
+
+    private LatencyAttribution workItemCreationLatency = null;
+    private final Map<State, Duration> aggregatedGetWorkStreamLatencies;
```

Comment:
nit: have a single Map from State -> sum/max
saves lookups but also makes it clearer the invariant that key is always in both maps

Reply:
Done.